### PR TITLE
alloc aligned memory for block bloom filter to work with jemalloc

### PR DESCRIPTION
**Pull request by dataroaring**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exprs/block_bloom_filter_impl.cc [hunk 1]
```diff
@@ -69,8 +72,12 @@ Status BlockBloomFilter::init_internal(const int log_space_bytes, uint32_t hash_
 
     const size_t alloc_size = directory_size();
     close(); // Ensure that any previously allocated memory for directory_ is released.
-    _mem_holder.reset(new char[alloc_size]);
```
Comment (xinyiZzz) [inline be/src/exprs/block_bloom_filter_impl.cc:72]: Remove definition of `_mem_holder`

File: be/src/exprs/block_bloom_filter_impl.cc [hunk 2]
```diff
@@ -69,8 +72,12 @@ Status BlockBloomFilter::init_internal(const int log_space_bytes, uint32_t hash_
...
-    _directory = (Bucket*)_mem_holder.get();
+    DCHECK(_directory == nullptr);
+    int rc = posix_memalign((void**)&_directory, 32, alloc_size);
```
Comment (xinyiZzz) [inline be/src/exprs/block_bloom_filter_impl.cc:76]: It seems that `posix_memalign` still allocates memory with `TCMalloc`, not `jemalloc`

**Comment by xinyiZzz** [comment]
LGTM @dataroaring rebase for p0 and UT

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**